azurecontainerapps.ts

```diff
@@ -232,9 +232,36 @@ export class azurecontainerapps {
     private static async getLocation(): Promise<string> {
         // Set deployment location, if provided
         let location: string = this.toolHelper.getInput('location', false);
+        let resourceGroup: string = this.toolHelper.getInput('resourceGroup', false);
+        let containerAppEnvironment: string = this.toolHelper.getInput('containerAppEnvironment', false);
 
-        // If no location was provided, use the default location for the Container App service
+        // If no location was provided, attempt to discover the location of the existing Container App Environment linked to the Container App
+        // or Container App Environment provided in the resource group or use the default location.
         if (this.util.isNullOrEmpty(location)) {
```

Comment:
I might have missed it, but if `location` is set, when do we use that value?
Also, please consider inverting your conditions and returning early, as else the code becomes nested and hard to follow. So for example, you could have:
```
if (!this.util.isNullOrEmpty(location)) {
    return location;
}

if (<next condition>) {
    executeXYZ
    return XYZ
}
```

Reply:
Ok, makes sense. Updated this.